src/main/java/net/solmis/cityparking/RateCalculator.java

```diff
@@ -0,0 +1,75 @@
+package net.solmis.cityparking;
+
+import net.solmis.cityparking.exceptions.CurrencyNotSupportedException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.Currency;
+
+@Component
+public class RateCalculator {
+
+    private static long centsForUnit = 100;
+    private static String defaultCurrencyCode;
+
+    @Value("${config.defaultCurrency}")
+    public void setDefaultCurrencyCode(String code) {
+        defaultCurrencyCode = code;
+    }
+
+    public static long calculateRegularCostInCents(long totalSeconds, Currency currency) throws CurrencyNotSupportedException {
+        long parkingHours = RateCalculator.toParkingHours(totalSeconds);
+        try {
+            double currencyFactor = RateCalculator.getCurrencyFactor(currency);
+            return calculateCost(parkingHours, currencyFactor, centsForUnit, 2*centsForUnit, 1.5);
+        } catch (CurrencyNotSupportedException e) {
+            throw e;
+        }
+    }
+
+    public static long calculateDisabledCostInCents(long totalSeconds, Currency currency) throws CurrencyNotSupportedException {
+        long parkingHours = RateCalculator.toParkingHours(totalSeconds);
+        try {
+            double currencyFactor = RateCalculator.getCurrencyFactor(currency);
+            return calculateCost(parkingHours, currencyFactor, 0, 2 * centsForUnit, 1.2);
+        } catch (CurrencyNotSupportedException e) {
```

Comment:
po co taki rethtrow?

Reply:
Mam świadomość, że to zło. Miałem nadzieję to później jakoś zmienić i ostatecznie niestety zostało. Pewnie lepiej było by już wcześniej sprawdzić czy ta waluta jest obsługiwana niż podnosić ten wyjątek coraz wyżej.